dialogy/plugins/text/classification/xlmr.py

```
@@ -75,97 +68,61 @@ def __init__(
-            self.load()
-            label_count = len(self.labelencoder.classes_)
-        if not label_count:
-            raise ValueError(
-                f"Plugin {self} needs either the training data "
-                "or an existing labelencoder to initialize."
-            )
+    def _request_model_inference(self, texts: List[str]) -> Tuple[List, List]:
+        payload = {"transcripts": texts}
 
         try:
-            self.model = self.classifier(
-                const.XLMR_MODEL,
-                self.model_dir,
-                num_labels=label_count,
-                use_cuda=self.use_cuda,
-                args=self.args_map,
-                **self.kwargs,
-            )
-        except OSError:
-            logger.info(f"Model not found at {self.model_dir}. "
-                        f"Default model weights will be loaded")
-            self.model = self.classifier(
-                const.XLMR_MODEL,
-                const.XLMR_MODEL_TIER,
-                num_labels=label_count,
-                use_cuda=self.use_cuda,
-                args=self.args_map,
-                **self.kwargs,
+            response = self.session.post(
+                self.url, json=payload, headers=self.headers, timeout=self.timeout
             )
 
-    @property
-    def valid_labelencoder(self) -> bool:
-        return hasattr(self.labelencoder, "classes_")
+            if response.status_code == 200:
+                # The API call was successful, expect the following to contain entities.
+                # A list of dicts or an empty list.
+                result = response.json()
+                return result.get("intents", []), result.get("logits", [])
+            
+        except requests.exceptions.Timeout as timeout_exception:
+            logger.error(f"Model Inference Service timed out: {timeout_exception}")  # pragma: no cover
+            logger.error(pformat(payload))  # pragma: no cover
+            return [], []  # pragma: no cover
```

This should also raise an exception?

---

Following duckling plugin conventions